src/server/zset_family_test.cc

```diff
@@ -757,6 +789,73 @@ TEST_F(ZSetFamilyTest, ZAddBug148) {
   EXPECT_THAT(resp, IntArg(1));
 }
 
+TEST_F(ZSetFamilyTest, ZMPopInvalidSyntax) {
+  // Not enough arguments.
+  auto resp = Run({"zmpop", "1", "a"});
+  EXPECT_THAT(resp, ErrArg("wrong number of arguments"));
+
+  // Zero keys.
+  resp = Run({"zmpop", "0", "MIN", "COUNT", "1"});
+  EXPECT_THAT(resp, ErrArg("syntax error"));
+
+  // Number of keys not uint.
+  resp = Run({"zmpop", "aa", "a", "MIN"});
+  EXPECT_THAT(resp, ErrArg("value is not an integer or out of range"));
+
+  // Missing MIN/MAX.
+  resp = Run({"zmpop", "1", "a", "COUNT", "1"});
+  EXPECT_THAT(resp, ErrArg("syntax error"));
+
+  // Wrong number of keys.
+  resp = Run({"zmpop", "1", "a", "b", "MAX"});
+  EXPECT_THAT(resp, ErrArg("syntax error"));
+
+  // Count with no number.
+  resp = Run({"zmpop", "1", "a", "MAX", "COUNT"});
+  EXPECT_THAT(resp, ErrArg("syntax error"));
+
+  // Count number is not uint.
+  resp = Run({"zmpop", "1", "a", "MIN", "COUNT", "boo"});
+  EXPECT_THAT(resp, ErrArg("value is out of range, must be positive"));
+}
+
+TEST_F(ZSetFamilyTest, ZMPop) {
+  // All sets are empty.
+  auto resp = Run({"zmpop", "1", "e", "MIN"});
+  EXPECT_THAT(resp, ArgType(RespExpr::NIL));
+
+  // Min operation.
+  resp = Run({"zadd", "a", "1", "a1", "2", "a2"});
+  EXPECT_THAT(resp, IntArg(2));
+
+  resp = Run({"zmpop", "1", "a", "MIN"});
+  EXPECT_THAT(resp, ContainsLabeledScoredArray("a", {{"a1", "1"}}));
+
+  // Max operation.
+  resp = Run({"zadd", "b", "1", "b1", "2", "b2"});
+  EXPECT_THAT(resp, IntArg(2));
+
+  resp = Run({"zmpop", "1", "b", "MAX"});
+  EXPECT_THAT(resp, ContainsLabeledScoredArray("b", {{"b2", "2"}}));
+
+  // Count > 1.
+  resp = Run({"zadd", "c", "1", "c1", "2", "c2"});
+  EXPECT_THAT(resp, IntArg(2));
+
+  resp = Run({"zmpop", "1", "c", "MAX", "COUNT", "2"});
+  EXPECT_THAT(resp, ContainsLabeledScoredArray("c", {{"c1", "1"}, {"c2", "2"}}));
+
+  // First non empty set is not the first set.
+  resp = Run({"zadd", "d", "1", "d1"});
+  EXPECT_THAT(resp, IntArg(1));
+
+  resp = Run({"zmpop", "2", "empty", "d", "MAX"});
+  EXPECT_THAT(resp, ContainsLabeledScoredArray("d", {{"d1", "1"}}));
+
+  // TODO: Add tests that check that sets that appear after the first non empty set are not
```

Comment:
yes:) and also please add to the test scenarios above a check that the element was actually popped, you can use the zcard command for this

Reply:
I used ZCARD to verify that sets are empty but ZRANGE for other cases since I wanted to verify that the correct elements are still there.